zipline/algorithm.py

```diff
@@ -510,15 +510,14 @@ def run(self, source, overwrite_sim_params=True,
                 self.sim_params.data_frequency,
             )
 
-        with ZiplineAPI(self):
-            # loop through simulated_trading, each iteration returns a
-            # perf dictionary
-            perfs = []
-            for perf in self.gen:
-                perfs.append(perf)
-
-            # convert perf dict to pandas dataframe
-            daily_stats = self._create_daily_stats(perfs)
+        # loop through simulated_trading, each iteration returns a
+        # perf dictionary
+        perfs = []
+        for perf in self.gen:
+            perfs.append(perf)
```

Comment:
A wrinkle here is that our internal code does not use this method, which is why I'm not looking at it this particular function too closely.

Though that adds another reason to the list for reasons to work on unification of 'how Zipline is run'.